diff -ruN a/src/libcrun/cgroup-resources.c b/src/libcrun/cgroup-resources.c
--- a/src/libcrun/cgroup-resources.c	2025-12-23 00:42:51.000000000 +0900
+++ b/src/libcrun/cgroup-resources.c	2026-03-09 16:54:14.396243240 +0900
@@ -179,15 +179,15 @@
       if (! found)
         {
           cleanup_free char *absolute_path = NULL;
-          libcrun_error_t tmp_err = NULL;
+          libcrun_error_t tmp_err2 = NULL;
 
           crun_error_release (err);
-          ret = get_realpath_to_file (cgroup_dirfd, "cgroup.controllers", &absolute_path, &tmp_err);
+          ret = get_realpath_to_file (cgroup_dirfd, "cgroup.controllers", &absolute_path, &tmp_err2);
           if (LIKELY (ret >= 0))
             ret = crun_make_error (err, 0, "controller `%s` is not available under %s", key, absolute_path);
           else
             {
-              crun_error_release (&tmp_err);
+              crun_error_release (&tmp_err2);
               ret = crun_make_error (err, 0, "the requested cgroup controller `%s` is not available", key);
             }
         }
@@ -598,7 +598,7 @@
       char fmt_buf_major[16];
       char fmt_buf_minor[16];
       char device[64];
-      int len;
+
 
 #define FMT_DEV(x, b)                                                                   \
   do                                                                                    \
@@ -859,7 +859,7 @@
       cleanup_free char *current = NULL;
       uint64_t limit = 0;
       uint64_t val, val_swap;
-      int ret;
+
 
       ret = read_all_file_at (dirfd, "memory.current", &current, NULL, err);
       if (UNLIKELY (ret < 0))
diff -ruN a/src/libcrun/container.c b/src/libcrun/container.c
--- a/src/libcrun/container.c	2025-12-23 00:42:51.000000000 +0900
+++ b/src/libcrun/container.c	2026-03-09 17:03:00.945025679 +0900
@@ -691,7 +691,7 @@
 
 static int
 do_hooks (runtime_spec_schema_config_schema *def, pid_t pid, const char *id, bool keep_going, const char *cwd,
-          const char *status, hook **hooks, size_t hooks_len, int out_fd, int err_fd, libcrun_error_t *err)
+          const char *status, hook **hook_entries, size_t hook_entries_len, int out_fd, int err_fd, libcrun_error_t *err)
 {
   size_t i, stdin_len;
   int r, ret;
@@ -801,22 +801,22 @@
 
   ret = 0;
 
-  for (i = 0; i < hooks_len; i++)
+  for (i = 0; i < hook_entries_len; i++)
     {
       char **env = environ;
 
-      if (hooks[i]->env)
-        env = hooks[i]->env;
+      if (hook_entries[i]->env)
+        env = hook_entries[i]->env;
 
-      ret = run_process_with_stdin_timeout_envp (hooks[i]->path, hooks[i]->args, cwd, hooks[i]->timeout, env,
+      ret = run_process_with_stdin_timeout_envp (hook_entries[i]->path, hook_entries[i]->args, cwd, hook_entries[i]->timeout, env,
                                                  stdin, stdin_len, out_fd, err_fd, err);
       if (UNLIKELY (ret != 0))
         {
           if (keep_going)
-            libcrun_warning ("error executing hook `%s` (exit code: %d)", hooks[i]->path, ret);
+            libcrun_warning ("error executing hook `%s` (exit code: %d)", hook_entries[i]->path, ret);
           else
             {
-              libcrun_error (0, "error executing hook `%s` (exit code: %d)", hooks[i]->path, ret);
+              libcrun_error (0, "error executing hook `%s` (exit code: %d)", hook_entries[i]->path, ret);
               break;
             }
         }
@@ -2201,7 +2201,7 @@
       if (UNLIKELY (terminal_fd_to < 0))
         return crun_make_error (err, errno, "dup terminal fd");
 
-      int i, non_blocking_fds[] = { terminal_fd_from, terminal_fd_to, 0, 1, -1 };
+      int non_blocking_fds[] = { terminal_fd_from, terminal_fd_to, 0, 1, -1 };
       for (i = 0; non_blocking_fds[i] >= 0; i++)
         {
           ret = set_blocking_fd (non_blocking_fds[i], false, err);
@@ -2234,7 +2234,7 @@
       struct epoll_event events[max_events];
       struct signalfd_siginfo si;
       struct winsize ws;
-      int i, nr_events;
+      int nr_events;
       ssize_t res;
 
       nr_events = TEMP_FAILURE_RETRY (epoll_wait (epollfd, events, max_events, -1));
@@ -4053,7 +4053,7 @@
 
       if (context->console_socket)
         {
-          int ret;
+
           cleanup_close int console_socket_fd = open_unix_domain_client_socket (context->console_socket, 0, err);
           if (UNLIKELY (console_socket_fd < 0))
             return console_socket_fd;
diff -ruN a/src/libcrun/container.h b/src/libcrun/container.h
--- a/src/libcrun/container.h	2025-12-23 00:42:51.000000000 +0900
+++ b/src/libcrun/container.h	2026-03-09 16:54:14.369349147 +0900
@@ -375,7 +375,7 @@
 
   if (ptr->linux.capabilities != NULL)
     {
-      for (size_t i = 0; ptr->linux.capabilities[i] != NULL; i++)
+      for (i = 0; ptr->linux.capabilities[i] != NULL; i++)
         free (ptr->linux.capabilities[i]);
       free (ptr->linux.capabilities);
     }
diff -ruN a/src/libcrun/ebpf.c b/src/libcrun/ebpf.c
--- a/src/libcrun/ebpf.c	2025-12-23 00:42:51.000000000 +0900
+++ b/src/libcrun/ebpf.c	2026-03-09 16:56:48.066579412 +0900
@@ -210,31 +210,31 @@
 
   if (has_type)
     {
-      struct bpf_insn i[] = { BPF_JMP_IMM (BPF_JNE, BPF_REG_2, bpf_type, number_instructions) };
+      struct bpf_insn insn[] = { BPF_JMP_IMM (BPF_JNE, BPF_REG_2, bpf_type, number_instructions) };
       number_instructions--;
-      program = bpf_program_append (program, i, sizeof (i));
+      program = bpf_program_append (program, insn, sizeof (insn));
     }
   if (has_access)
     {
-      struct bpf_insn i[] = {
+      struct bpf_insn insn[] = {
         BPF_MOV32_REG (BPF_REG_1, BPF_REG_3),
         BPF_ALU32_IMM (BPF_AND, BPF_REG_1, bpf_access),
         BPF_JMP_REG (BPF_JNE, BPF_REG_1, BPF_REG_3, number_instructions - 2),
       };
       number_instructions -= 3;
-      program = bpf_program_append (program, i, sizeof (i));
+      program = bpf_program_append (program, insn, sizeof (insn));
     }
   if (has_major)
     {
-      struct bpf_insn i[] = { BPF_JMP_IMM (BPF_JNE, BPF_REG_4, major, number_instructions) };
+      struct bpf_insn insn[] = { BPF_JMP_IMM (BPF_JNE, BPF_REG_4, major, number_instructions) };
       number_instructions--;
-      program = bpf_program_append (program, i, sizeof (i));
+      program = bpf_program_append (program, insn, sizeof (insn));
     }
   if (has_minor)
     {
-      struct bpf_insn i[] = { BPF_JMP_IMM (BPF_JNE, BPF_REG_5, minor, number_instructions) };
+      struct bpf_insn insn[] = { BPF_JMP_IMM (BPF_JNE, BPF_REG_5, minor, number_instructions) };
       number_instructions--;
-      program = bpf_program_append (program, i, sizeof (i));
+      program = bpf_program_append (program, insn, sizeof (insn));
     }
 
   if (has_type == 0 && has_access == 0 && has_major == 0 && has_minor == 0)
@@ -255,7 +255,7 @@
 bpf_program_complete_dev (struct bpf_program *program, libcrun_error_t *err arg_unused)
 {
 #ifdef HAVE_EBPF
-  struct bpf_insn i[] = {
+  struct bpf_insn insn[] = {
     BPF_MOV64_IMM (BPF_REG_0, 0),
     BPF_EXIT_INSN (),
   };
@@ -263,7 +263,7 @@
   if (program->private & HAS_WILDCARD)
     return program;
 
-  program = bpf_program_append (program, &i, sizeof (i));
+  program = bpf_program_append (program, &insn, sizeof (insn));
 #endif
   return program;
 }
diff -ruN a/src/libcrun/linux.c b/src/libcrun/linux.c
--- a/src/libcrun/linux.c	2025-12-23 00:42:51.000000000 +0900
+++ b/src/libcrun/linux.c	2026-03-09 16:54:14.419295317 +0900
@@ -373,8 +373,8 @@
 
       if (relative)
         {
-          runtime_spec_schema_defs_id_mapping **mappings;
-          size_t mappings_len;
+          runtime_spec_schema_defs_id_mapping **id_mappings;
+          size_t id_mappings_len;
 
           if (def == NULL
               || def->linux == NULL
@@ -382,17 +382,17 @@
               || (! is_uids && def->linux->gid_mappings_len == 0))
             return crun_make_error (err, 0, "specified a relative mapping without user namespace mappings");
 
-          mappings_len = (is_uids ? def->linux->uid_mappings_len : def->linux->gid_mappings_len);
-          mappings = is_uids ? def->linux->uid_mappings : def->linux->gid_mappings;
+          id_mappings_len = (is_uids ? def->linux->uid_mappings_len : def->linux->gid_mappings_len);
+          id_mappings = is_uids ? def->linux->uid_mappings : def->linux->gid_mappings;
 
-          for (i = 0; i < mappings_len; i++)
-            if (value[1] >= mappings[i]->container_id && value[1] < mappings[i]->container_id + mappings[i]->size)
+          for (i = 0; i < id_mappings_len; i++)
+            if (value[1] >= id_mappings[i]->container_id && value[1] < id_mappings[i]->container_id + id_mappings[i]->size)
               break;
 
-          if (i == mappings_len)
+          if (i == id_mappings_len)
             return crun_make_error (err, 0, "could not find a user namespace mapping for the relative mapping `%s`", option);
 
-          value[1] += mappings[i]->host_id - mappings[i]->container_id;
+          value[1] += id_mappings[i]->host_id - id_mappings[i]->container_id;
         }
       if (written > allocated - 64)
         {
@@ -1599,17 +1599,17 @@
 
   if (binds)
     {
-      cleanup_close int fd = -1;
+      cleanup_close int fd_inner = -1;
 
       if (rel_dev)
         {
-          fd = openat (devfd, rel_dev, O_NOFOLLOW | O_CLOEXEC | O_PATH | O_NONBLOCK);
-          if (UNLIKELY (fd < 0))
+          fd_inner = openat (devfd, rel_dev, O_NOFOLLOW | O_CLOEXEC | O_PATH | O_NONBLOCK);
+          if (UNLIKELY (fd_inner < 0))
             {
               if (errno == ENOENT)
-                fd = openat (devfd, rel_dev, O_CREAT | O_NOFOLLOW | O_CLOEXEC | O_NONBLOCK, 0700);
+                fd_inner = openat (devfd, rel_dev, O_CREAT | O_NOFOLLOW | O_CLOEXEC | O_NONBLOCK, 0700);
 
-              if (UNLIKELY (fd < 0))
+              if (UNLIKELY (fd_inner < 0))
                 return crun_make_error (err, errno, "openat `%s`", device->path);
             }
         }
@@ -1617,19 +1617,19 @@
         {
           const char *rel_path = consume_slashes (normalized_path);
 
-          fd = crun_safe_create_and_open_ref_at (false, get_private_data (container)->rootfsfd, rootfs, rel_path, 0755, err);
-          if (UNLIKELY (fd < 0))
-            return fd;
+          fd_inner = crun_safe_create_and_open_ref_at (false, get_private_data (container)->rootfsfd, rootfs, rel_path, 0755, err);
+          if (UNLIKELY (fd_inner < 0))
+            return fd_inner;
         }
 
       if (srcfd >= 0)
         {
-          ret = syscall_move_mount (srcfd, "", fd, "", MOVE_MOUNT_T_EMPTY_PATH | MOVE_MOUNT_F_EMPTY_PATH);
+          ret = syscall_move_mount (srcfd, "", fd_inner, "", MOVE_MOUNT_T_EMPTY_PATH | MOVE_MOUNT_F_EMPTY_PATH);
           if (LIKELY (ret >= 0))
             return 0;
         }
 
-      ret = do_mount (container, fullname, fd, normalized_path, NULL, MS_BIND | MS_PRIVATE | MS_NOEXEC | MS_NOSUID, NULL, LABEL_MOUNT, err);
+      ret = do_mount (container, fullname, fd_inner, normalized_path, NULL, MS_BIND | MS_PRIVATE | MS_NOEXEC | MS_NOSUID, NULL, LABEL_MOUNT, err);
       if (UNLIKELY (ret < 0))
         return ret;
     }
